puppet/manifests/default.pp

```diff
@@ -1,3 +1,38 @@
+class glusterfs_common {
+  # This contains setup common to all gluster nodes. Peers and volumes need to
+  # be configured on the individual nodes.
+
+  # NOTE: If there are three gluster nodes, an extra puppet provisioning run
+  # needs to happen after all nodes are up and running so that the existing
+  # cluster can invite new nodes. I have no idea what happens with four or
+  # more.
+
+  # Vagrant gets very sad if gluster can't find the peer it's looking for, so
+  # we need to wrap the `gluster` command and pretend `gluster peer probe`
+  # succeeded even if it didn't.
+  $gluster_wrapper = "#!/bin/bash
+gluster \"$@\"
+CODE=$?
+if [ \"\$1 \$2\" = \"peer probe\" ]; then CODE=0; fi
```

Comment:
We could probably work around this by writing our own gluster module (or forking and modifying the one we're using), but that seems like a bunch of work. :-/